### PR TITLE
Corrects syntax in S3 commandlets to always separate bucket and key prefix

### PR DESCRIPTION
**Pull request by lorengordon**
The S3 commandlets or the S3 API have changed, such that the -BucketName argument only works when it is just the bucket. It no longer supports when the argument also includes the KeyPath… E.g. This fails now:

```
Write-S3Object -BucketName terrafirm/key/prefix/path -File file.log
```

But this succeeds:

```
Write-S3Object -BucketName terrafirm -Key key/prefix/path/file.log -File file.log
```

### CODE REVIEW

**Comment by lorengordon** [comment]
go codebuild go

**Comment by lorengordon** [comment]
go codebuild go

File: templates/win_userdata.ps1 [hunk 1]
```diff
@@ -32,13 +36,14 @@ cd $TempDir
...
+    [Parameter(Mandatory = $false)][string]$Msg
   )
 
-  $DebugFile = "$TempDir\debug.log"
+  $DebugFileName = "debug.log"
+  $DebugFile = "$TempDir\$DebugFileName"
   "$(Get-Date): $Msg" | Out-File $DebugFile -Append -Encoding utf8
-  Write-S3Object -BucketName "$BuildSlug/$BuildLabel" -File $DebugFile
-  Write-S3Object -BucketName "$BuildSlug/$BuildLabel" -File $UserdataLogFile
+  Write-S3Object -BucketName "$BuildBucket" -Key "$${BuildKeyPrefix}/$${BuildLabel}/$${DebugFileName}" -File "$DebugFile"
```
[the remark this answers is not in the document]
Comment (lorengordon) [reply templates/win_userdata.ps1:45]: The file is templated using terraform's `templatefile()`, and that templating also uses `${}` to substitute variables. The escape for that is to double the dollar sign.

In bash, using `${}` helps when using interpolation to concatenate strings, as we're doing here, combining many variables with the `/` path separator.

So in the end, after the templating, bash needs to see `${}`, but it's getting processed by `templatefile()` first, which means we need to escape `${}` anywhere it's _not_ a terraform variable....

**Review by eemperor — APPROVED** [review]

File: templates/win_userdata.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (eemperor) [reply templates/win_userdata.ps1:45]: Got it.  Thanks for that explanation.

**Pull request merged**